### ISSUE TITLE
SemVer issues with 1.2.3

### ISSUE THREAD

**Issue by petetnt** [rexxars/react-markdown#14]
I wrote this here, but it is tied with `react-commonmark-renderer` too.

The update to the `react-commonmark-renderer` which updated the `commonmark` version itself included breaking changes which are somewhat incompatible with semver, which causes issues for those who have locked up the version of `react-markdown`. For example, my package was locked `react-markdown@1.2.0` due to #11. But as the `react-markdown` and `react-commonmark-renderer` are using the caret notation in `package.json`, it's possible for `npm@3` to resolve the dependencies to both `react-markdown` and `react-commonmark-renderer` in a way that are incompatible, thus introducing breaking changes.

1. `npm install react-markdown@v1.2.0`
2. Render a markdown file that would respond to `Header` in the old version (´Heading` in the new version)

Expected: 
  -  Should render a Header element 

Actual:
  - Throws for unknown nodeType `Header`

The structure looks like this:
``` 
node_modules
  | - react-markdown@1.2.0
  | - react-commonmark-renderer@2.2.2
  | - commonmark@0.22.1
```

As the [npm docs state](https://docs.npmjs.com/misc/semver#caret-ranges-1-2-3-0-2-5-0-0-4) 

> #### Caret Ranges ^1.2.3 ^0.2.5 ^0.0.4

> Allows changes that do not modify the left-most non-zero digit in the [major, minor, patch] tuple. In other words, this allows patch and minor updates for versions 1.0.0 and above, patch updates for versions 0.X >=0.1.0, and no updates for versions 0.0.X.

> Many authors treat a 0.x version as if the x were the major "breaking-change" indicator.

> Caret ranges are ideal when an author may make breaking changes between 0.2.4 and 0.3.0 releases, which is a common practice. However, it presumes that there will not be breaking changes between 0.2.4 and 0.2.5. It allows for changes that are presumed to be additive (but non-breaking), according to commonly observed practices.

As the requirement has changed from `0.22.0` to `0.24.0`, npm picks up the newest version of `0.22.x` instead -> `0.22.1`.

**Comment by rexxars** [comment]
Yes, I was careless when I merged the pull request that changed this, thinking I had covered the backwards-incompatible changes by some code internally. I'll be more careful with this going forward, rather bumping the major version in cases where there is any chance of breakage, even if I think I've covered most cases.

I'm a little unsure what to do at this time - I'm thinking it might be best to roll back to the previous version of commonmark and release it as 1.2.4, then releasing a 2.0.0 which uses the new version of commonmark. Does that sound alright?
